### PR TITLE
Solutions to programming problems with test cases

### PR DESCRIPTION
**Pull request by santosh-m**

### CODE REVIEW

File: Solutions/BrianCarName.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	/**
+	 * Checks if given car name is liked by Brian
+	 * 
+	 * @param carName the given car name
+	 * @return the string
+	 */
+	public String checkCarName(String carName) {
+		if (carName != null  && !carName.isEmpty() && !carName.trim().isEmpty()) {
+			logger.log(Level.INFO, "Input String/Car Name : " + carName);
+			TreeMap<Character,Integer> charOccurrenceMap = populateMap(carName);
```
Comment (neerajkujha) [inline Solutions/BrianCarName.java:33]: is treemap really required here?

File: Test_Cases/AlicePrimeStringTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		Assert.assertEquals("Error", testAlicePrimeString.checkPrimeString(""));
+		Assert.assertEquals("Error", testAlicePrimeString.checkPrimeString(" "));
+		Assert.assertEquals("Error", testAlicePrimeString.checkPrimeString(null));
+	}
+
+	/**
+	 * Test CheckPrimeString verify logs.
+	 */
+	@Test
+	public void testCheckPrimeStringVerifyLogs() {
```
Comment (neerajkujha) [inline Test_Cases/AlicePrimeStringTest.java:60]: how this test is verifying logs?

File: Solutions/BrianCarName.java (hunk 1 above)
Comment (santosh-m) [reply Solutions/BrianCarName.java:33]: @neerajkujha Yes TreeMap is required, since I'm using `charOccurrenceMap.firstEntry()` further in the solution. i.e `firstEntry` api is available only for TreeMap but not in case of HashMap or LinkedHashMap. But then yes, solution is possible by using HashMap as well, in that case i would need to convert values of map to an array & get its 1st element to achieve same as above.

File: Test_Cases/AlicePrimeStringTest.java (hunk 2 above)
Comment (santosh-m) [reply Test_Cases/AlicePrimeStringTest.java:60]: @neerajkujha This test case is similar to positive test case, it's asserting expected value(String) with actual value. Just that this test case is meant for verifying logs manually by the developer. If we need to verify logs programmatically/using JUnit, one way of doing it is, asserting the count of `log` statements of `Level.INFO` & `Level.SEVERE`

File: Solutions/BrianCarName.java (hunk 1 above)
Comment (neerajkujha) [reply Solutions/BrianCarName.java:33]: Good, but treemap keeps key in sorted order, do you think we need the keys in order here. Also do we really need firstentry

File: Test_Cases/AlicePrimeStringTest.java (hunk 2 above)
Comment (neerajkujha) [reply Test_Cases/AlicePrimeStringTest.java:60]: hmm, but logs can also be verified with the test cases you have written for positive and negative scenario. So what is the utility of just one positive test case in test logs. or do the above test cases not print the logs?

File: Solutions/BrianCarName.java (hunk 1 above)
Comment (santosh-m) [reply Solutions/BrianCarName.java:33]: The only purpose I've used `TreeMap` here is for `firstEntry` api. The reason being I could stop validating further for character (C3) occurrence, in a scenario where 2nd character (C2) occurrence is not equal to 1st character (C1) occurrence. Yes I agree, we could use other simpler approach like putting C1, C2, C3 occurrences in `HashSet` & validating if it's size is equal to one. I had chosen former approach thinking we could stop validating an iteration before, in the above scenario.

File: Test_Cases/AlicePrimeStringTest.java (hunk 2 above)
Comment (santosh-m) [reply Test_Cases/AlicePrimeStringTest.java:60]: Yes, even positive & negative test case do print logs. I believed that it would be easier to verify the logs with a dedicated test case for this purpose, since it invokes actual method to test only once. In case of positive & negative test case it has multiple invocations of the method, & it would be a little cumbersome to verify the logs. Yes I agree this test case utility isn't that significant.
